pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/LegacyKitResampleRequest.java

```diff
@@ -0,0 +1,92 @@
+
+@Getter
+@AllArgsConstructor
+/**
+ * Request to resample a kit for a participant with a legacy short ID
+ * The request is for resampling a kit from Pepper Guid and Hruid to their old legacy ids
+ */
+public class LegacyKitResampleRequest {
+    String currentCollaboratorSampleId;
+    String newCollaboratorSampleId;
+    String newCollaboratorParticipantId;
+    String shortId;
+    String legacyShortId;
+
+    /**
+     * Verify that the request is valid,
+     * that all the fields are present,
+     * that the participant exists in the ES with the given short ID,
+     * that the given legacy short ID matches the one on file,
+     * that the kit request exists for the given collaborator sample ID to resmaple,
+     * and also that the new collaborator sample ID does not already exist
+     *
+     * @param ddpInstance DDP instance for the request
+     * @param kitRequestDao DAO for kit requests
+     */
+    public void verify(DDPInstance ddpInstance, KitRequestDao kitRequestDao) {
+        if (ddpInstance == null) {
+            throw new DsmInternalError("DDP instance not found");
+        }
+        checkNotEmptyRequestFields();
+        Map<String, Map<String, Object>> esParticipantData = ElasticSearchUtil.getSingleParticipantFromES(ddpInstance.getName(),
+                ddpInstance.getParticipantIndexES(), shortId);
+        if (esParticipantData.size() != 1) {
+            throw new DSMBadRequestException("Invalid participant short ID " + shortId);
+        }
+        Map<String, Object> ptp = esParticipantData.values().stream().findFirst().orElseThrow();
+        Map<String, Object> profile = (Map<String, Object>) ptp.get("profile");
+        if (profile == null) {
+            throw new DSMBadRequestException("No profile found for participant short ID " + shortId);
+        }
+        String legacyShortIdOnFile = profile.get("legacyShortId").toString();
+        if (!legacyShortId.equals(legacyShortIdOnFile)) {
+            throw new DSMBadRequestException(("Legacy short ID %s does not match legacy short ID on file %s for participant short ID %s, "
+                    + " will not resample kit %s").formatted(legacyShortId, legacyShortIdOnFile, shortId, currentCollaboratorSampleId));
+        }
+        if (!kitRequestDao.existsKitRequestWithCollaboratorSampleId(currentCollaboratorSampleId, (String) profile.get("guid"))) {
+            throw new DSMBadRequestException("Kit request not found for collaborator sample ID %s".formatted(currentCollaboratorSampleId));
+        }
+        if (kitRequestDao.existsKitRequestWithCollaboratorSampleId(newCollaboratorSampleId, (String) profile.get("guid"))) {
+            throw new DSMBadRequestException("Kit request with the new collaboratorSampleId %s already exists!"
+                    .formatted(newCollaboratorSampleId));
+        }
+
+    }
+
+    /**
+     * Check that all required fields are present in the request
+     */
+    private void checkNotEmptyRequestFields() {
```

Comment:
Can this method be eliminated by using `final` fields?  I'm not familiar with the options for `@AllArgsConstructor`, but one thing that could help here is to remove the annotation and then make a constructor that does all of these checks so that it's impossible to create a new object with invalid fields.